### PR TITLE
Using otto to solve orientation problem

### PR DESCRIPTION
**Pull request by javiergamarra**
Recreated to refactor some code

This is a hard-to-explain pull request.

The current code (and the previous async task code) has a bug/problem with orientation changes. When the screen rotates, the activity gets destroyed, creating leaks, bugs...

With this PR these problems are fixed because when the thread finishes, either someone is listening to the event (the activity or the new activity are alive) and receives the event or noone listens.

This solution could be improved (in a later PR) sending tasks in an executor (instead a thread) and wrapping the runnable to allow passing parameters.

Although otto is deprecated (eventbus also has problems), the trendy alternative, RxJava, is hard to explain in a PR :P

### CODE REVIEW

**Comment by mdelapenya** [comment]
Merge conflicts on MainActivity.java, could you rebase please?

**Comment by javiergamarra** [comment]
done :)

File: app/src/main/java/es/craftsmanship/toledo/katangapp/utils/AndroidBus.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public static Bus getInstance() {
+        return BUS;
+    }
+
+    public static class MainThreadBus extends Bus {
+        private final Handler handler = new Handler(Looper.getMainLooper());
+
+        @Override
+        public void post(final Object event) {
+            if (Looper.myLooper() == Looper.getMainLooper()) {
```
Comment (mdelapenya) [inline app/src/main/java/es/craftsmanship/toledo/katangapp/utils/AndroidBus.java:27]: equals()?

Comment (javiergamarra) [inline app/src/main/java/es/craftsmanship/toledo/katangapp/utils/AndroidBus.java:27]: It's equivalent. We are checking to see if we are already executing code on the main thread so if the reference (unique) is the same, we are and we don't have to post a message with the handler.

Comment (mdelapenya) [inline app/src/main/java/es/craftsmanship/toledo/katangapp/utils/AndroidBus.java:27]: Ok, so it's about references, I'll keep it as you did ;)

**Pull request merged**